erpnext/manufacturing/doctype/production_plan/production_plan.py

```diff
@@ -599,13 +607,13 @@ def get_sales_orders(self):
 			and so.docstatus = 1 and so.status not in ("Stopped", "Closed")
 			and so.company = %(company)s
 			and so_item.qty > so_item.work_order_qty {0} {1}
-			and (exists (select name from `tabBOM` bom where bom.item=so_item.item_code
+			and (exists (select name from `tabBOM` bom where bom.item= {2}
 					and bom.is_active = 1)
 				or exists (select name from `tabPacked Item` pi
 					where pi.parent = so.name and pi.parent_item = so_item.item_code
 						and exists (select name from `tabBOM` bom where bom.item=pi.item_code
 							and bom.is_active = 1)))
-		""".format(so_filter, item_filter), {
+		""".format(so_filter, item_filter, variant_of_bom or "so_item.item_code"), {
```

Comment:
avoid using `format` for queries you are constructing without ORM, it is not the safest option plus escaping issues. Instead of `{2}` use `%(bom_item)s` and pass the values in dict such as `{"bom_item" : value, "item_filter": value}`

Reply:
used frappe.db.escape instead